File: v1/ecmascript/phone-number/phone-number.js
export default class PhoneNumber {
    constructor(input) {
        this.validatedNumber = input.replace(/(\D)/g, '');

        if (this.validatedNumber.length === 11 && this.validatedNumber[0] === '1') {
            this.validatedNumber = this.validatedNumber.substr(1);
        }

        if (this.validatedNumber.length !== 10) {
            this.validatedNumber = '0000000000';
        }
    }

    number() {
        return this.validatedNumber;
    }

    areaCode() {
        return this.validatedNumber.substr(0, 3);
    }

    toString() {
        return `(${this.areaCode()}) ${this.validatedNumber.substr(3, 3)}-${this.validatedNumber.substr(6, 4)}`;
    }
}

File: v1/kotlin/phone-number/src/main/kotlin/PhoneNumber.kt
class PhoneNumber(input: String) {
    val invalidNumber = "0000000000"

    val number: String
    val areaCode: String
    val middle: String
    val end: String

    init {
        val digits = input.filter { Character.isDigit(it) }

        if (digits.length == 10)
            number = digits
        else if (digits.length == 11 && digits[0] == '1')
            number = digits.slice(1 until digits.length)
        else
            number = invalidNumber

        areaCode = number.slice(0 .. 2)
        middle = number.slice(3 .. 5)
        end = number.slice(6 .. 9)
    }

    override fun toString(): String = "($areaCode) $middle-$end"
}